### PR TITLE
Move all the patch-level hist vars to site-level

### LINKED ISSUE #475: remove patch level history output?
Correct me if I'm wrong, but I don't think there is any reason for us to save history variables at the patch level.  When we pass them to the land model, the land model performs the area weighting and scales those values up to the grid-level anyway.  

Our site level diagnostics are essentially doing the same thing, but it is more transpartent to the fates user when we do the upscaling to the site/column level on our side.  Either way, we should not be achieving the same result two different ways.

### PR DESCRIPTION
**Pull request by ckoven**
There have been a few variables that stayed on the old patch native weighting in the history writing.  The use of these is confusing for a few reasons: its confusingly inconsistent with how we're doing things otherwise, its more complicated (since the patch weighting system weighted by patch canopy area rather than patch area for reasons having to do with radiative transfer and the existence of a "bare-ground" patch in that scheme), and it won't play well with more complex subgrid structure once we start moving to using other-than-natural-veg land types.  So this PR gets rid of all of the remaining ones (those that aren't already removed in #640).

problem noted in #475 and #630.  fixes #475.

### Collaborators:
discussed with @rgknox, @rosiealice, and @glemieux 

### Expectation of Answer Changes:
should be bit for bit or roundoff-level changes only, with the exception that I got rid of one redundant variable, `NPP_column`.

### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [ ] If answers were expected to change, evaluation was performed and provided

### Test Results:
not yet run through test suite.

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

File: main/FatesHistoryInterfaceMod.F90 [hunk 1]
```diff
@@ -3132,36 +3104,26 @@ subroutine update_history_prod(this,nc,nsites,sites,dt_tstep)
             ccohort => cpatch%shortest
             do while(associated(ccohort))
                
-               ! TODO: we need a standardized logical function on this (used lots, RGK)
```
Comment (glemieux) [inline main/FatesHistoryInterfaceMod.F90:3135]: Nice, this takes care of the TODO.

File: main/FatesHistoryInterfaceMod.F90 [hunk 2]
```diff
@@ -3592,10 +3553,8 @@ subroutine update_history_hydraulics(this,nc,nsites,sites,bc_in,dt_tstep)
                
                ! TODO: we need a standardized logical function on this (used lots, RGK)
                if ((cpatch%area .gt. 0._r8) .and. (cpatch%total_canopy_area .gt. 0._r8)) then
```
[the remark this answers is not in the document]
Comment (ckoven) [reply main/FatesHistoryInterfaceMod.F90:3555]: yes we can

[the remark this answers is not in the document]
Comment (ckoven) [reply main/FatesHistoryInterfaceMod.F90:3555]: actually we ca just get rid of the whole clause.  the n_density was only there for legacy reasons i think and the n_perm2 isn't used.

File: main/FatesHistoryInterfaceMod.F90 [hunk 3]
```diff
@@ -3592,10 +3553,8 @@ subroutine update_history_hydraulics(this,nc,nsites,sites,bc_in,dt_tstep)
...
-                  n_density = ccohort%n/min(cpatch%area,cpatch%total_canopy_area) 
                   n_perm2   = ccohort%n/AREA
```
[the remark this answers is not in the document]
Comment (glemieux) [reply main/FatesHistoryInterfaceMod.F90]: Resolving per above comment.

File: main/FatesHistoryInterfaceMod.F90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (glemieux) [reply main/FatesHistoryInterfaceMod.F90]: Good point!

**Comment by glemieux** [comment]
All expected PASS:
```
/glade/u/home/glemieux/scratch/clmed-tests/pr642-nopa-hist-vars-C242d6d14-Fc65614ee.fates.cheyenne.intel
/glade/u/home/glemieux/scratch/clmed-tests/pr642-nopa-hist-vars-C242d6d14-Fc65614ee.fates.cheyenne.gnu
```
All DIFFs related to the changes from patch to site-level.  @ckoven reviewed and thinks the differences are reasonable, noting that it likely has to do with the "time-averaging differences in patch areas".

**Comment by ckoven** [comment]
at least i think so...

**Comment by rgknox** [comment]
I like how more lines were removed than were added.  I think this looks good.

**Comment by glemieux** [comment]
Status update: ran a quick test late yesterday after the deconflict merge and a bunch of tests are failing during run time.  Looking into the cause.

**Comment by glemieux** [comment]
> Status update: ran a quick test late yesterday after the deconflict merge and a bunch of tests are failing during run time. Looking into the cause.

Problem turned out to be the fact that #640 updated the `vname` for three of the fire variables (see https://github.com/NGEET/fates/commit/5ac1cebe44b441253df4bd8a5f4ba7c97764aebc).  I assumed the capitalization wouldn't matter, but the user namelist variables need to have the same case.  Lesson learned.  CTSM PR created to correct this:  https://github.com/ESCOMP/CTSM/pull/1011

**Comment by glemieux** [comment]
Re-test against current master baseline results in the same DIFFs as seen previously.  All expected PASS.
```
/glade/u/home/glemieux/scratch/clmed-tests/sci.1.35.3_api.10.0.1-clm5.0.30-C95df92ee-F8259b5b2.fates.cheyenne.intel
/glade/u/home/glemieux/scratch/clmed-tests/sci.1.35.3_api.10.0.1-clm5.0.30-C95df92ee-F8259b5b2.fates.cheyenne.gnu
```

**Pull request merged**